client/web/src/global/CoolCodeIntel.tsx

```diff
@@ -0,0 +1,842 @@
+    } & Omit<BlobProps, 'className' | 'wrapCode' | 'blobInfo' | 'disableStatusBar'>
+> = props => {
+    const { data, error, loading } = useQuery<CoolCodeIntelReferencesResult, CoolCodeIntelReferencesVariables>(
+        FETCH_REFERENCES_QUERY,
+        {
+            variables: {
+                repository: props.clickedToken.repoName,
+                commit: props.clickedToken.commitID,
+                path: props.clickedToken.filePath,
+                // ATTENTION: Off by one ahead!!!!
+                line: props.clickedToken.line - 1,
+                character: props.clickedToken.character - 1,
+                after: null,
+                filter: props.filter || null,
+            },
+            // Cache this data but always re-request it in the background when we revisit
+            // this page to pick up newer changes.
+            fetchPolicy: 'cache-and-network',
+            nextFetchPolicy: 'network-only',
+        }
+    )
+
+    // If we're loading and haven't received any data yet
+    if (loading && !data) {
+        return (
+            <>
+                <LoadingSpinner inline={false} className="mx-auto my-4" />
+                <p className="text-muted text-center">
+                    <i>Loading references ...</i>
+                </p>
+            </>
+        )
+    }
+
+    // If we received an error before we had received any data
+    if (error && !data) {
+        throw new Error(error.message)
+    }
+
+    // If there weren't any errors and we just didn't receive any data
+    if (!data || !data.repository?.commit?.blob?.lsif) {
+        return <>Nothing found</>
+    }
+
+    const lsif = data.repository?.commit?.blob?.lsif
+
+    return (
+        <SideReferencesLists
+            {...props}
+            references={lsif.references}
+            definitions={lsif.definitions}
+            implementations={lsif.implementations}
+            hover={lsif.hover}
+        />
+    )
+}
+
+interface LSIFLocationResult {
+    __typename?: 'LocationConnection'
+    nodes: ({ __typename?: 'Location' } & LocationFields)[]
+    pageInfo: { __typename?: 'PageInfo'; endCursor: Maybe<string> }
+}
+
+export const SideReferencesLists: React.FunctionComponent<
+    {
+        clickedToken: CoolClickedToken
+        setActiveLocation: (location: Location | undefined) => void
+        activeLocation: Location | undefined
+        filter: string | undefined
+        references: LSIFLocationResult
+        definitions: Omit<LSIFLocationResult, 'pageInfo'>
+        implementations: LSIFLocationResult
+        hover: Maybe<{
+            __typename?: 'Hover'
+            markdown: { __typename?: 'Markdown'; html: string; text: string }
+        }>
+    } & Omit<BlobProps, 'className' | 'wrapCode' | 'blobInfo' | 'disableStatusBar'>
+> = props => {
+    const { references, definitions, implementations, hover } = props
+    const references_: Location[] = useMemo(() => references.nodes.map(buildLocation), [references])
```

Comment:
Are these type annotations required? 

```suggestion
    const references_ = useMemo(() => references.nodes.map(buildLocation), [references])
```